app/services/depositor.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Completes the deposit of an object
+class Depositor
+  class NotDepositingError < StandardError; end
+
+  def self.deposit(object_version:)
+    new(object_version: object_version).deposit
+  end
+
+  attr_reader :object_version
+
+  def initialize(object_version:)
+    @object_version = object_version
+  end
+
+  def deposit
+    verify_depositing!
+
+    parent.event_context = {
+      user: sdr_user,
+      description: "What changed: #{what_changed}"
+    }
+
+    object_version.deposit_complete!
+  end
+
+  def parent
+    # Though object and object.head.collection/object.head.work are the same in DB, they are not the same in memory.
+    # Thus, need to set event_context on the traversed parent object.
+    object_version.try(:collection) || object_version.work
+  end
+
+  private
+
+  def sdr_user
+    User.find_by!(name: 'SDR')
+  end
+
+  def what_changed
+    object_version.version_description.presence || 'not specified'
+  end
+
+  def verify_depositing!
+    return if object_version&.depositing?
+
+    raise NotDepositingError, "Cannot deposit because state is not `depositing` (`#{object_version.state}`)"
```

Comment:
This error message is confusing. The object is already deposited.

Reply:
@justinlittman I'm open to suggestions. FWIW, as currently written, this exception message should never be surfaced in the app, as it's caught by the containing context.

Reply:
I'm dumping the exception altogether.